### DISCUSSION TITLE
How to choose the editor different then $EDITOR for current file opening?

### DISCUSSION THREAD

**Discussion by AdamJel** [sayanarijit/xplr#587] [Q&A]
I have my default editor defined, but what if I want to open the file in a different one just this one time? Is there a way how to trigger editor selection? Or perhaps a second keybinding `:e2`?

Thank you

**Comment by sayanarijit** [comment]
Yes. You can override the key binding, or add a new one. See the docs, there are examples there. You can also use the map.xplr plugin to run custom commands on any file/list of files.

**Reply by AdamJel** [reply]
I am trying to add a new keybinding for action mode `:g`, which would open `$EDITOR_GUI` application. But I have problems figuring out the correct format. I started with copying `:e` keybinding, which in default `init.lua` is:

```lua
2015      ["e"] = {
2016        help = "open in editor",
2017        messages = {
2018          {
2019            BashExec0 = [===[
2020              ${EDITOR:-vi} "${XPLR_FOCUS_PATH:?}"
2021            ]===],
2022          },
2023          "PopMode",
2024        },
2025      },
```

And edited it as:
```lua
      ["g"] = {
        help = "open in GUI editor",
        messages = {
          {
            BashExec0 = [===[
              ${EDITOR_GUI} "${XPLR_FOCUS_PATH:?}"
            ]===],
          },
          "PopMode",          
        },
      },
```

I do see the entry in `Help [action to]`, but when I press <kbd>g</kbd>, I get this error: `ERROR: process exited with code exit status: 126`.

I suppose there is some silly error you @sayanarijit will spot at once?
Thank you

**Reply by sayanarijit** [reply]
You can spawn a shell by typing :! And run the script there to see what's wrong.
